### PR TITLE
Find all paths when no branch specified

### PR DESCRIPTION
**Pull request by hughsaunders**

### CODE REVIEW

**Comment by sigmavirus24** [comment]
Looks like a good minor refactor to pull some code into a reusable function. :+1:

File: scripts/upgrade-branches.sh [hunk 1]
```diff
@@ -16,88 +16,103 @@ function check_tag {
...
+  elif [[ $next_minor_version != '' ]]; then
+    test_branches="$test_branches minor-to-$next_minor_version"
+  fi
+
+  echo $test_branches
+}
+
+# --- Main ---
+branch="$1"
+if [[ -z "$branch" ]]; then
```
Comment (git-harry) [inline scripts/upgrade-branches.sh]: I think the output from the script should always be in the same format. I'd suggest moving the for-loop out of the conditional and have branches be a list of one branch if branch is already defined.

Comment (hughsaunders) [inline scripts/upgrade-branches.sh]: Done

File: scripts/upgrade-branches.sh [hunk 2]
```diff
@@ -16,88 +16,103 @@ function check_tag {
...
+  branches=$(git branch -a |awk -F'[/ ]' '/[a-z]+-[0-9]+\.[0-9]+$/{print $NF}' |sort -u)
```
Comment (git-harry) [inline scripts/upgrade-branches.sh]: This doesn't include master

Comment (hughsaunders) [inline scripts/upgrade-branches.sh]: Done

File: scripts/upgrade-branches.sh [hunk 3]
```diff
@@ -16,88 +16,103 @@ function check_tag {
...
+    test_branches="$test_branches minor-to-$next_minor_version"
+  fi
+
+  echo $test_branches
+}
+
+# --- Main ---
+branch="$1"
+if [[ -z "$branch" ]]; then
+  branches="master $(git branch -a |awk -F'[/ ]' '/[a-z]+-[0-9]+\.[0-9]+$/{print $NF}' |sort -u)"
```
Comment (git-harry) [inline scripts/upgrade-branches.sh]: I fetched all the remote branches and now the scripts is outputting bug branches, the following should fix it.
```
git branch -a |awk -F'[/ ]' '/\/[a-z]+-[0-9]+\.[0-9]+$/{print $NF}' |sort -u
```

Comment (hughsaunders) [inline scripts/upgrade-branches.sh]: Good spot, I hadn't fetched all branches in my test repo. Done

**Comment by sigmavirus24** [comment]
:+1:

**Pull request merged**